### PR TITLE
Forward commit commands for local transactions as well as XA transact…

### PR DESCRIPTION
**Pull request by jgallimore**
…ions during ACK compaction.

This PR will forward all commits with transaction info (as opposed to just XA transactions, which would not include commits for local transactions). I have included a unit test, and will open a discussion on the mailing list.

### CODE REVIEW

File: activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/MessageDatabase.java [hunk 1]
```diff
@@ -2236,7 +2236,7 @@ private boolean shouldForward(JournalCommand<?> command) {
                 result = true;
             } else if (command instanceof KahaCommitCommand) {
                 KahaCommitCommand kahaCommitCommand = (KahaCommitCommand) command;
-                if (kahaCommitCommand.hasTransactionInfo() && kahaCommitCommand.getTransactionInfo().hasXaTransactionId()) {
+                if (kahaCommitCommand.hasTransactionInfo())  {
```
Comment (gtully) [inline activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/MessageDatabase.java:2239]: we don't need that check for hasTransactionInfo any more, it was redundant before but is more so now. But it is not a biggie.

Comment (jgallimore) [reply activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/MessageDatabase.java:2239]: That does make sense - I guess we wouldn't see a commit command without transaction info. I'll remove that check.

File: activemq-unit-tests/src/test/java/org/apache/activemq/bugs/AMQ7067Test.java [hunk 2]
```diff
@@ -526,6 +526,82 @@ public boolean isSatisified() throws Exception {
         }
     }
 
+    @Test
+    public void testForwardAcksAndCommitsWithLocalTransaction() throws Exception {
```
Comment (gtully) [inline activemq-unit-tests/src/test/java/org/apache/activemq/bugs/AMQ7067Test.java:530]: great test :-)

**Review by gtully — APPROVED** [review]

File: activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/MessageDatabase.java (hunk 1 above)
Comment (gtully) [reply activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/MessageDatabase.java]: thanks. less is always more.

**Comment by jgallimore** [comment]
I've (hopefully) simplified that shouldForward() method, per the comments. Thank you for the review, @gtully !

**Pull request merged**